pkg/config/config.go

```diff
@@ -150,3 +156,27 @@ func Finalize(pArgs *ProgArgs) error {
 	}
 	return err
 }
+
+func ValidateConfigRootPath(configRoot string) (string, error) {
+	allowedPrefixes := []string{
+		"/etc/rte",
+		"/run/rte",
+		"/var/rte",
+		"/usr/local/etc/rte",
+		"/home", // mostly for tests
+	}
+	if configRoot == "" {
+		return configRoot, fmt.Errorf("configRoot is not allowed to be an empty string")
+	}
+	// Resolve and clean the input path
+	absPath, err := filepath.Abs(filepath.Clean(configRoot))
+	if err != nil {
+		return configRoot, fmt.Errorf("failed to validate configRoot path: %w", err)
+	}
+	for _, prefix := range allowedPrefixes {
+		if strings.HasPrefix(absPath, prefix) {
+			return absPath, nil
```

Comment:
if the prefix is matching, let's not return the original path (even though is correct and we know it) but let's build a path and return it re-creating the user-supplied path (!!).
This should make the tool happy. I hope.